ouroboros-network/src/Ouroboros/Network/Subscription/Common.hs

```diff
@@ -202,16 +202,28 @@ subscribeTo tbl threadPool tracer localIPv4_m localIPv6_m connectionAttemptDelay
                         else return ConnectFail
         traceWith tracer $ SubscriptionTraceConnectEnd remoteAddr result
         case result of
-            ConnectSuccess -> k sd -- TODO catch all application exception.
+            ConnectSuccess -> runApplication sd
             ConnectSuccessLast -> do
                     outstandingConThreads <- atomically $ readTVar conThreads
                     mapM_ (\a -> throwTo a
                             (SubscriberError SubscriberParrallelConnectionCancelled
                             "Parrallel connection cancelled"
                             callStack)) outstandingConThreads
-                    k sd
+                    runApplication sd
             ConnectFail -> return ()
 
+    runApplication :: Socket.Socket -> IO ()
+    runApplication sd = do
+        r_e <- try (k sd)
```

Comment:
I would be inclined to use `catch` instead of `try` as we don't need to do anything in case no exception is thrown.

For example:
```haskell
k sd `catch` \e -> do
  traceWith tracer $ SubscriptionApplicationException e
  -- TODO: Fatal/Permanent errors such as filesystem corruption should be
  -- permitted to trickle up instead of just causing the creation of a new
  -- connection.
  throwM e
```

Reply:
I'll switch to `catch`.